### PR TITLE
refactor(pattern): added data-autoid to optional fields and it's parent

### PR DESCRIPTION
**Pull request by sannminwin**
### Related Ticket(s)

#637 Add data-autoid attributes to optional fields for Patterns

### Description
This pull request adds data-autoid to optional fields and it's parent element.

### Changelog

**New**

- added data-autoid attribute to optional fields and it's parent element of  patterns.

### CODE REVIEW

File: packages/patterns-react/src/patterns/LeadSpace/LeadSpace.js [hunk 1]
```diff
@@ -97,7 +97,9 @@ const LeadSpace = ({
     <section
       data-autoid={`${stablePrefix}--leadspace`}
       className={className(variation, theme)}>
-      <div className={`${prefix}--leadspace__container`}>
+      <div
+        data-autoid={`${stablePrefix}--leadspace__container`}
```
Comment (annawen1) [inline packages/patterns-react/src/patterns/LeadSpace/LeadSpace.js]: This here seems kinda excessive to me, we already have an autoid in for the `section` element above

Comment (sannminwin) [reply packages/patterns-react/src/patterns/LeadSpace/LeadSpace.js]: Yeah, I see your point. We already have it at parent level. And I just talked with Ke and confirmed that they need `data-autoid` on all optional fields and it's immediate parent.

**Comment by jeffchew** [comment]
@sannminwin you probably need to update the corresponding READMEs too, right?

File: packages/patterns-react/src/patterns/LeadSpaceCentered/LeadSpaceCentered.js [hunk 2]
```diff
@@ -98,12 +98,23 @@ const LeadSpaceCentered = ({
...
-          <p className={`${prefix}--leadspace--centered__desc`}>{copy}</p>
-          {buttons && buttons.length > 0 && <ButtonGroup buttons={buttons} />}
+          <p
+            data-autoid={`${stablePrefix}--leadspace--centered__desc`}
+            className={`${prefix}--leadspace--centered__desc`}>
+            {copy}
+          </p>
+          {buttons && buttons.length > 0 && (
+            <ButtonGroup
+              data-autoid={`${stablePrefix}--leadspace--centered__button-group`}
```
Comment (annawen1) [inline packages/patterns-react/src/patterns/LeadSpaceCentered/LeadSpaceCentered.js]: I was going to do this a bit later but can you edit `ButtonGroup` to accept this prop? 😁 It doesn't actually set the data id here. Probably pass it in as `autoid` prop and then in `ButtonGroup` you set `data-autoid` there.

File: packages/patterns-react/src/patterns/LeadSpace/LeadSpaceButtons.js [hunk 3]
```diff
@@ -40,7 +40,6 @@ const ButtonGroup = ({ buttons }) => {
     const button2 = buttonsRef.current[1];
 
     if (button1.current.offsetWidth != button2.current.offsetWidth) {
-      console.log('triggered');
```
Comment (annawen1) [inline packages/patterns-react/src/patterns/LeadSpace/LeadSpaceButtons.js]: Thanks, that's been hanging out there for a while...😅

File: packages/patterns-react/src/patterns/LeadSpaceCentered/LeadSpaceCentered.js (hunk 2 above)
Comment (sannminwin) [reply packages/patterns-react/src/patterns/LeadSpaceCentered/LeadSpaceCentered.js]: The end goal is to add `autoid` to each button and it's parent.
It seems I have to work on ButtonGroup.

Comment (annawen1) [reply packages/patterns-react/src/patterns/LeadSpaceCentered/LeadSpaceCentered.js]: Right, yeah you can pass in the base part of the autoid then.

```
<ButtonGroup autoid={`${stablePrefix}--leadspace--centered`) ..... />
```

and then in ButtonGroup.js:
```
const ButtonGroup = ({ buttons, autoid }) => {
  return featureFlag(
    DDS_BUTTON_GROUP,
    <ol
      className={`${prefix}--buttongroup`}
      data-autoid={`${autoid}--button-group`}>
      {buttons.map((button, key) => {
        return (
          <li key={key} className={`${prefix}--buttongroup-item`}>
            <Button
              data-autoid={`${autoid}--button-group-${key}`}
              {...button}
              kind={key === 0 ? 'primary' : 'tertiary'}>
              {button.copy}
            </Button>
          </li>
        );
      })}
    </ol>
  );
};
```

**Pull request merged**